Description: Fixes the compatibility with the version of xpp2 in Debian
Origin: backport, https://github.com/dom4j/dom4j/pull/22/files
--- a/src/main/java/org/dom4j/xpp/ProxyXmlStartTag.java
+++ b/src/main/java/org/dom4j/xpp/ProxyXmlStartTag.java
@@ -205,14 +205,31 @@
         }
     }
 
-    @Override
-    public boolean removeAttributeByName(String s, String s1) throws XmlPullParserException {
-        throw new UnsupportedOperationException();
+    public boolean removeAttributeByName(String namespaceURI, String localName) throws XmlPullParserException {
+        if (element != null) {
+            for (Iterator<Attribute> iter = element.attributeIterator(); iter.hasNext();) {
+                Attribute attribute = iter.next();
+
+                if (namespaceURI.equals(attribute.getNamespaceURI())
+                        && localName.equals(attribute.getName())) {
+                    return element.remove(attribute);
+                }
+            }
+        }
+        return false;
     }
 
-    @Override
-    public boolean removeAttributeByRawName(String s) throws XmlPullParserException {
-        throw new UnsupportedOperationException();
+    public boolean removeAttributeByRawName(String rawName) throws XmlPullParserException {
+        if (element != null) {
+            for (Iterator<Attribute> iter = element.attributeIterator(); iter.hasNext();) {
+                Attribute attribute = iter.next();
+
+                if (rawName.equals(attribute.getQualifiedName())) {
+                    return element.remove(attribute);
+                }
+            }
+        }
+        return false;
     }
 
     /**
